drivers/soundwire/intel.c

```diff
@@ -1293,6 +1293,9 @@ static int intel_link_probe(struct auxiliary_device *auxdev,
 	/* use generic bandwidth allocation algorithm */
 	sdw->cdns.bus.compute_params = sdw_compute_params;
 
+	/* avoid resuming from pm_runtime suspend if it's not required */
+	dev_pm_set_driver_flags(dev, DPM_FLAG_SMART_SUSPEND);
```

Comment:
@plbossart this is a nice find but I wonder if we still need the checks for if (pm_runtime_suspended(dev)) in some places in the prepare/suspend callbacks for the auxdev if you were to use smart suspend

Reply:
@plbossart the documentation for the SMART_SUSPEND flags states this: "Setting that flag causes the PM core and middle-layer code (bus types, PM domains etc.) to skip the ->suspend_late and ->suspend_noirq callbacks provided by the driver if the device remains in runtime suspend throughout those phases of the system-wide suspend (and similarly for the “freeze” and “poweroff” parts of system hibernation). [Otherwise the same driver callback might be executed twice in a row for the same device, which would not be valid in general.] If the middle-layer system-wide PM callbacks are present for the device then they are responsible for skipping these driver callbacks; if not then the PM core skips them. The subsystem callback routines can determine whether they need to skip the driver callbacks by testing the return value from the dev_pm_skip_suspend() helper function."
I dont think this mean that this will avoid a runtime resume during suspend right?

Reply:
also if we're use SMART_SUSPEND, do we also need SKIP_RESUME?

Reply:
I suppose the documentation is obsolete or incomplete then